test/test-PreviewProvider.js

```diff
@@ -50,6 +50,19 @@ const gMockMetadataStore = {
       });
     }
     return items;
+  },
+  asyncExecuteQuery(query) {
+    // extract key from the query
+    let key = query.split("=")[1].trim().replace(/'/g, "");
```

Comment:
As i kind of mentioned above, ```asyncGetMetadataByCacheKey``` does a lot of extra work that we don't actually need here, whereas doing the query itself is quicker in this case. This line in this test (i know i'm not a fan either) is a work around to extract the key to shim it. Alternatively, to make this a little more robust, I can extract the query string and  construct it once as a default option, and then override it during test time.
What do you think?